<!--
 * @Author: Gatsby
 * @Date: 2021-03-26 11:17:27
 * @LastEditTime: 2021-03-26 11:24:51
 * @LastEditors: Gatsby
 * @Description: 
 * @FilePath: \ES6\catch方法.html
 * 可以输入预定的版权声明、个性签名、空行等
-->
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>
    <script>
        const p = new Promise((resolve, regect) => {
            setTimeout(() => {
                //设置 p 对象状态为失败，并设置失败的值
                regect("Error !!!");
            }, 1000)
        });

        // p.then(function (value) {

        // }, function (reason) {
        //     console.error(reason);
        // });

        p.catch(function (reason) { 
            console.warn(reason);
         });
         
    </script>
</body>

</html>